test/Feature/RangeSpec.hs

```diff
@@ -181,20 +176,41 @@ spec = do
             [json|[{"id":1},{"id":2},{"id":3},{"id":4},{"id":5},{"id":6},{"id":7},{"id":8},{"id":9},{"id":10},{"id":11},{"id":12},{"id":13},{"id":14},{"id":15}]|]
             { matchHeaders = ["Content-Range" <:> "0-14/*"] }
 
-      it "fails if limit equals 0" $
+      it "returns blank array when limit is 0" $
         get "/items?select=id&limit=0"
-          `shouldRespondWith` [json|{"message":"HTTP Range error"}|]
-          { matchStatus  = 416
+          `shouldRespondWith` [json|[]|]
+          { matchStatus  = 200
           , matchHeaders = [matchContentTypeJson]
           }
 
       it "fails if limit is negative" $
-        get "/items?select=id&limit=-1"
-          `shouldRespondWith` [json|{"message":"HTTP Range error"}|]
+        get "/clients?select=id,projects(id,tasks(id))&order=id.asc&limit=-1&projects.order=id.asc&projects.limit=1&projects.tasks.order=id.asc&projects.tasks.limit=1" `shouldRespondWith`
```

Comment:
@gautam1168 Sorry for the late review. This looks almost done now.

Q: Why do we need the embeds here? We already have some embeds tests below and seems this test would be clearer without them.

Reply:
Yes these are not needed. I just used the same endpoint for al lthe tests. But I will change this to the previous version.

Reply:
Reverted the test to use the non-embedded resource for testing as it was before